### PR TITLE
refactor: replace manage:user with Permissions

### PR DESCRIPTION
**Pull request by miketheman**
Anywhere we had `manage:user`, use a more specific Permissions entry.

A couple of non-Account-related permissions were added here, as the
previous use of `manage:user` was put in place of a more specific
permission, since that's all we had.

Signed-off-by: Mike Fiedler <miketheman@gmail.com>

### CODE REVIEW

File: warehouse/config.py [hunk 1]
```diff
@@ -135,7 +135,21 @@ class RootFactory:
...
-        (Allow, Authenticated, "manage:user"),
+        (
+            Allow,
+            Authenticated,
+            (
+                Permissions.Account2FA,
+                Permissions.AccountAPITokens,
+                Permissions.AccountManage,
+                Permissions.AccountManagePublishing,
+                Permissions.AccountVerifyEmail,
```
Comment (ewdurbin) [inline warehouse/config.py:146]: just a thought, but some of these feel a bit too granular but overall seem reasonable to me. i think the general question i'd ask when granularizing is "would another entity do this on the behalf of another". in this instance i think that most of the Verify* will only be acted on by a user so in my (very negotiable) mental model we could just have "Permissions.AccountVerify" that covers all of these.

probably no action here just writing down the thought provoked.

**Review by ewdurbin — APPROVED** [review]
forward! i really dig the outcomes here and am excited to see us closer to the end goal.

File: tests/unit/macaroons/test_security_policy.py [hunk 2]
```diff
@@ -316,7 +317,7 @@ def test_permits_valid_macaroon(self, monkeypatch, principals, expected):
 
     @pytest.mark.parametrize(
         "invalid_permission",
-        ["admin", "moderator", "manage:user", "manage:project", "nonexistent"],
```
[the remark this answers is not in the document]
Comment (miketheman) [reply tests/unit/macaroons/test_security_policy.py:319]: That's correct - `moderator` is gone as well. I expect to remove `manage:project` when I refactor it to oblivion.

File: warehouse/config.py (hunk 1 above)
Comment (miketheman) [reply warehouse/config.py:146]: I hear you - I was going through and breaking things down to the smallest unit first, and then seeing where things can maybe be combined. I'll keep these in mind when I take a more holistic view of account provisioning, and may bundle many of them together then.

**Pull request merged**